### DISCUSSION TITLE
Change Working Group meeting times?

### DISCUSSION THREAD

**Discussion by leebyron** [graphql/graphql-wg#1051] [Polls]
> # UPDATE
>
> After fairly broad support for introducing twice-weekly meetings, along with other feedback, we will be implementing a change to WG meeting times.
>
> Those changes are detailed in this PR: https://github.com/graphql/graphql-wg/pull/1125
>
> The important changes relative to proposals discussed within this thread are:
>
> - The existing WG meeting is now named "Primary", remains on first Thursdays, but is reduced to 1.5hr and shifted slightly earlier.
> - Two new "Secondary" WG meetings are added, one at the same time on third Thursdays, and one +5hr on second Wednesdays to be more APAC available.

---

Our last couple working group meetings have gone a bit over time where we've needed to shift topics to subsequent meetings or async discussions.

These meetings used to be THREE hours (😓) which we reduced to two hours after feedback that these were too long to stay focused and a bit more challenging to schedule, especially across multiple time zones.

I'd love to pull feedback on whether we should consider expanding. I'll list a few proposals here with an open poll, but please feel free to add ideas as well.

**Option A: Business as usual**

Continue to use our 2hr session as scheduled, and shift this discussion to how to make better use of the time we set up, rather than extending time (eg. more attention to async discussions, additional breakout meetings, etc)

**Option B: Return to three hours**

Expand the time to 3 hours with an ability to break early if agenda runs short. To address meeting fatigue, we would take 5m breaks at the end of every hour.

**Option C: Next day overflow**

Keep a 2hr session, however book the same time of day time slot for the following day (Thursday & Fridays), providing a total of *four* hours available to discuss. If the agenda is completed in the first session then we cancel the second.

**Option D: Same day overflow**

Keep a 2hr session, and include an additional 2hr session later the same day. This may require shifting times a bit, and we would need to be more sensitive about timezones of discussion, but this may introduce an opportunity to be more inclusive of Asia/Pacific WG members. Meetings could be morning and afternoon Pacific America time which would map across [timezones](https://www.timeanddate.com/worldclock/meetingtime.html?iso=20220707&p1=224&p2=179&p3=268&p4=240) to be evening Europe time for the first and morning Asia time for the second.

**Option E: Twice monthly meetings**

We could move to twice monthly meetings. Rather than having an explicit overflow, we would just have a more frequent cadence so the cost of delaying an agenda item until the following meeting is less painful. (h/t @twof )

**Comment by leebyron** [comment]
CC: @graphql/tsc, @andimarek, and other recent attendees @adron @twof @asprouse @benjie @BoD @IvanGoncharov @n1rual @mjmahone @michaelstaib @ndejaco2 @robgrant @robrichard @rivantsov @saihaj @sachee @LunaticMuch @timsuchanek @yaacovCR

**Comment by yaacovCR** [comment]
I voted for next day overflow because I liked the idea, but time zone/day wise I would not be able to attend the second session (my Friday evening). We could implement the same idea but just have the two days be Wednesday/Thursday.

or whatever works for the most folks

**Reply by benjie** [reply]
I also cannot easily do Fridays in general (but I did not vote for this option).

**Reply by leebyron** [reply]
Wed/Thu instead of Thu/Fri is definitely an option to explore

**Comment by twof** [comment]
Would moving to 1.5hr biweekly meetings at the same time be easier to manage for folks?

**Reply by leebyron** [reply]
I added this to the poll list, which frustratingly deleted all votes placed so far

**Reply by twof** [reply]
Thanks!

**Reply by LunaticMuch** [reply]
> Would moving to 1.5hr biweekly meetings at the same time be easier to manage for folks?

Why not even weekly of one hour? 
A weekly of a fixed, but shorter time would work better IMO. You get used to that and also your personal/work diary is more manageable on a long run.
Also if you can't make, you do not lose many discussion points. You don't have the need to setup a break every 50 mins and, finally, there's no uncontrolled overflow risk.

**Comment by leebyron** [comment]
I'll bring this back to live discussion in today's meeting, but for those wanting to pre-read, here's a concrete proposal I'll introduce and we'll discuss from there:

Main changes:

* Change day of week to Wednesdays
* Increase frequency to weekly meetings
* Reduce time per meeting from 2hr to 1.5hr
* Alternate weekly between EU-friendly times and AP-friendly times.

Goals:

* Introduces a regular time slot to better support an Asia/Pacific contributing audience
* For our current Europe-centric audience, increases frequency and improves time.

Time Proposal:

* Even-numbered ISO-weeks: [Wednesdays @ 9:30am - 11am PT](https://www.timeanddate.com/worldclock/fixedtime.html?msg=Example+US%2FEU+GraphQL+WG&iso=20220810T0930&p1=224&ah=1&am=30)
  * 12:30pm for NYC, 5:30pm London, 6:30pm Zurich
* Odd-numbered ISO-weeks: [Wednesdays @ 3:30pm - 5pm PT](https://www.timeanddate.com/worldclock/fixedtime.html?msg=Example+US%2FEU+GraphQL+WG&iso=20220817T1530&p1=224&ah=1&am=30)
  * 6:30pm for NYC, 8:30am for Sydney

**Reply by mjmahone** [reply]
Not super important to figure out during the WG, but just logistically we need to address.

We'll need to either choose one of these to "officially" be the monthly TSC meeting (probably can just keep being the first-of-the-month meeting), or we need to change the TSC doc to saying you must attend one of the last 3 *weekly* (rather than monthly) meetings to count for quorum.

I'm fine with expecting TSC members to attend at least one WG meeting every 3-weeks-to-month. But regardless just a flag.

**Reply by benjie** [reply]
Really good point. I've proposed some edits to the TSC document to keep expectations roughly the same independent of how frequently the WG meets (so long as it is at least monthly): https://github.com/graphql/graphql-wg/pull/1098

**Reply by benjie** [reply]
With the move to weekly meetings; can I also propose that we disallow people adding topics to the next meeting? This would mean that every topic added to an agenda would have at least 1 week of notice so that people who aren't in the optimal timezone can make arrangements in order to be able to attend topics that they're interested in.

**Reply by dondonz** [reply]
Thanks so much for introducing an Australia-friendly meeting time!

**Reply by andimarek** [reply]
When does this new schedule start?

**Reply by LunaticMuch** [reply]
@benjie can you give all a message then new meetings are in the calendar so we can refresh and check?  for some reason, today's meeting has disappeared from the shared calendar (at least on mine!)

**Reply by benjie** [reply]
@LunaticMuch Are you referring to the composite schema's WG in 5 hours time? It does seem to still be in the foundation calendar: https://calendar.google.com/calendar/u/0/embed?src=linuxfoundation.org_ik79t9uuj2p32i3r203dgv5mo8@group.calendar.google.com

**Reply by LunaticMuch** [reply]
Yes. For some reason it disappeared from my sync. Need to forcely removed the calendar and add it back.

**Reply by dondonz** [reply]
I would be super keen on the new Asia Pacific-friendly meeting option. Will the new meeting schedule be starting soon?

**Reply by leebyron** [reply]
I've got one last iteration of the schedule I'm working on that includes updates to the full suite of GraphQL Foundation meetings - my goal is to get them shipped for October!

**Reply by dondonz** [reply]
Thanks @leebyron!

I'm mindful there's plenty of work behind the scenes to make the new meeting cadence happen, if there's any way I can help out please let me know!

**Reply by leebyron** [reply]
Thanks everyone.

The final version of this proposal is detailed in this PR: https://github.com/graphql/graphql-wg/pull/1125
